exasol_integration_test_docker_environment/test/test_test_container_reuse.py

```diff
@@ -0,0 +1,169 @@
+                no_test_container_cleanup_after_success=True,
+                no_test_container_cleanup_after_failure=False
+            )
+        test_container_future_1 = yield from self.run_dependencies(test_container_task_1)
+        container_info = test_container_future_1.get_output()  # type: ContainerInfo
+        container = self._client.containers.get(container_info.container_name)
+
+        self.return_object({"container_id": container.image.id, "image_id": container.image.id})
+
+
+class TestContainerReuseTest(unittest.TestCase):
+
+    def set_job_id(self, task_cls):
+        strftime = datetime.now().strftime('%Y_%m_%d_%H_%M_%S')
+        job_id = f"{strftime}_{task_cls.__name__}"
+        config = luigi.configuration.get_config()
+        config.set('job_config', 'job_id', job_id)
+        return job_id
+
+    def clean(self):
+        jobid = self.set_job_id(CleanImagesStartingWith)
+        task = CleanImagesStartingWith(starts_with_pattern="", jobid=jobid)
+        luigi.build([task], workers=1, local_scheduler=True, log_level="INFO")
+        if task._get_tmp_path_for_job().exists():
+            shutil.rmtree(str(task._get_tmp_path_for_job()))
+
+    def setUp(self):
+        self.client = docker.from_env()
+        self.temp_directory = tempfile.mkdtemp()
+        resource_directory = Path(Path(__file__).parent, "resources/test_test_container_reuse")
+        self.working_directory = shutil.copytree(resource_directory,
+                                                 Path(self.temp_directory, "test_test_container_reuse"))
+        os.chdir(self.working_directory)
+        set_build_config(force_rebuild=False,
+                         force_pull=False,
+                         force_rebuild_from=tuple(),
+                         log_build_context_content=False,
+                         output_directory=self.temp_directory,
+                         cache_directory="",
+                         build_name="",
+                         temporary_base_directory="/tmp"
+                         )
+        set_docker_repository_config(
+            docker_password=None,
+            docker_repository_name=self.__class__.__name__,
+            docker_username=None,
+            tag_prefix="",
+            kind="source"
+        )
+        self.clean()
+
+    def tearDown(self):
+        self.clean()
+        shutil.rmtree(self.temp_directory)
+        self.client.close()
+
+    def run1(self):
+        try:
+            self.set_job_id(SpawnTestContainer)
+            task = TestTask(reuse=False, attempt=1)
+            success = luigi.build([task], workers=1, local_scheduler=True, log_level="INFO")
+            if success:
+                result = task.get_return_object()
+                return result
+            else:
+                Exception("Task failed")
+        finally:
+            if task._get_tmp_path_for_job().exists():
+                shutil.rmtree(str(task._get_tmp_path_for_job()))
```

Comment:
Same as above, tearDown() calls shutil.rmtree(self.temp_directory), will that not remove this directory, too?